firestore/integration_test_internal/src/source_test.cc

```diff
@@ -0,0 +1,523 @@
+    errorPromise.set_value(error_code);
+  });
+  errorFuture.wait();
+
+  Future<QuerySnapshot> future = colRef.Get(Source::kCache);
+  WaitFor(future);
+  QuerySnapshot querySnapshot = *future.result();
+  ASSERT_TRUE(querySnapshot.metadata().is_from_cache());
+  ASSERT_TRUE(querySnapshot.metadata().has_pending_writes());
+  ASSERT_EQ(4, querySnapshot.DocumentChanges().size());
+  std::map<std::string, MapFieldValue> newData{
+      {"doc1", {{"key1", FieldValue::String("value1")}}},
+      {"doc2",
+       {{"key2", FieldValue::String("value2")},
+        {"key2b", FieldValue::String("value2b")}}},
+      {"doc3", {{"key3b", FieldValue::String("value3b")}}},
+      {"doc4", {{"key4", FieldValue::String("value4")}}},
+  };
+  ASSERT_EQ(newData, QuerySnapshotToMap(querySnapshot));
+
+  future = colRef.Get();
+  WaitFor(future);
+  querySnapshot = *future.result();
+  ASSERT_TRUE(querySnapshot.metadata().is_from_cache());
+  ASSERT_TRUE(querySnapshot.metadata().has_pending_writes());
+  ASSERT_EQ(4, querySnapshot.DocumentChanges().size());
+  ASSERT_EQ(newData, QuerySnapshotToMap(querySnapshot));
+
+  future = colRef.Get(Source::kServer);
+  Await(future);
+  EXPECT_EQ(future.status(), FutureStatus::kFutureStatusComplete);
+  EXPECT_EQ(future.error(), Error::kErrorUnavailable);
+}
+
+TEST_F(SourceTest, TestGetNonExistingDocWhileOnlineWithDefaultGetOptions) {
+  DocumentReference docRef = Document();
+
+  Future<DocumentSnapshot> future = docRef.Get();
+  WaitFor(future);
+
+  DocumentSnapshot snapshot = *future.result();
+  ASSERT_FALSE(snapshot.exists());
+  ASSERT_FALSE(snapshot.metadata().is_from_cache());
+  ASSERT_FALSE(snapshot.metadata().has_pending_writes());
+}
+
+TEST_F(SourceTest,
+       TestGetNonExistingCollectionWhileOnlineWithDefaultGetOptions) {
+  CollectionReference colRef = Collection();
+
+  Future<QuerySnapshot> future = colRef.Get();
+  WaitFor(future);
+
+  QuerySnapshot querySnapshot = *future.result();
+  ASSERT_TRUE(querySnapshot.empty());
+  ASSERT_EQ(0, querySnapshot.DocumentChanges().size());
+  ASSERT_FALSE(querySnapshot.metadata().is_from_cache());
+  ASSERT_FALSE(querySnapshot.metadata().has_pending_writes());
+}
+
+TEST_F(SourceTest, TestGetNonExistingDocWhileOfflineWithDefaultGetOptions) {
+  DocumentReference docRef = Document();
+
+  WaitFor(TestFirestore()->DisableNetwork());
+  Future<DocumentSnapshot> future = docRef.Get();
+  Await(future);
+  EXPECT_EQ(future.status(), FutureStatus::kFutureStatusComplete);
+  EXPECT_EQ(future.error(), Error::kErrorUnavailable);
+}
+
+TEST_F(SourceTest, TestGetDeletedDocWhileOfflineWithDefaultGetOptions) {
+  DocumentReference docRef = Document();
+  WaitFor(docRef.Delete());
+
+  WaitFor(TestFirestore()->DisableNetwork());
+  Future<DocumentSnapshot> future = docRef.Get();
+  WaitFor(future);
+
+  DocumentSnapshot snapshot = *future.result();
+  ASSERT_FALSE(snapshot.exists());
+  ASSERT_EQ(0, snapshot.GetData().size());
+  // TODO(ehsann): is_from_cache() is false. why?
```

Comment:
It looks like there's a related TODO in the Android test -- can you copy that comment here and mark the test as disabled (it's `@Ignore`d on Android)? (To disable a test, add `DISABLED_` to its name, i.e., it should be `DISABLED_GetDeletedDocWhileOfflineWithDefaultGetOptions`).

Reply:
Done